src/uk/gov/hmcts/ardoq/ArdoqClient.groovy

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ardoq
+
+import uk.gov.hmcts.contino.RepositoryUrl
+
+class ArdoqClient {
+  String apiKey
+  String apiUrl
+  public steps
+
+  ArdoqClient(apiKey, apiUrl, steps) {
+    this.apiKey = apiKey
+    this.apiUrl = apiUrl
+    this.steps = steps
+  }
+
+  String getApplicationId() {
+    def applicationId = this.steps.env.ARDOQ_APPLICATION_ID
+    if (!applicationId?.trim()) {
+      this.steps.echo "Ardoq Application Id is not configured for ${this.steps.env.GIT_URL}"
+      return
+    }
+    return applicationId?.trim();
+  }
+
+  String getRepositoryName() {
+    return new RepositoryUrl().getShortWithoutOrgOrSuffix(this.steps.env.GIT_URL)
+  }
+
+  String getLanguage() {
```

Comment:
might be better to pass this in or subclass per language rather than trying to detect as it should probably be unneeded? but no harm for now

Reply:
I think in order to get the specific version this seemed like the most reliable way and just happens to work for both Java and Nodejs.
You're right though, as this grows to other ares of the business to support non-dockerised repos I'll need to refactor this.